src/jit/lower.cpp

```diff
@@ -3685,6 +3685,9 @@ void Lowering::DoPhase()
         {
             TreeNodeInfoInit(node);
 
+            // Only nodes that produce values should have a non-zero dstCount.
+            assert((node->gtLsraInfo.dstCount == 0) || node->IsValue());
+
```

Comment:
`X || Y` will be true if either `X` or `Y` is true, but we want the exclusive or of `X` and `Y`. You could write `X != Y` but 1) I don't think it is as clear to the reader (and this is, after all, an assert that runs only in Debug or Checked mode), and 2) it would rely on the conditionals returning the same value for `true`, which I don't think is always the case.